ocaml/libs/xapi-stdext/lib/xapi-stdext-unix/unixext.ml

```diff
@@ -586,6 +585,55 @@ let with_polly_wait kind fd f =
       in
       f wait fd
 
+module Timeout = struct
+  type t = Mtime.Span.t
```

Comment:
I think these modules need a comment what the main abstraction here is. A timeout is a time span?

Reply:
I don't understand why this module is needed at all. When using time, I've never had to convert from and to string, other than logging. If a conversion from a string is needed, first convert it to float, or from an uint64, and use a conversion that clearly states the units. Having conversion functions with implicit units opens possibility to misuse and should be avoided.

Reply:
The timeout is present in a configuration file, so we need to convert from a string, and due to how config file parsing is implemented in XAPI we must always provide a conversion to a string too.

Reply:
We could potentially move this somewhere else that deals with configuration files (although both xapi and xenopsd need it)

Reply:
> use a conversion that clearly states the units

I could require to always include `s` at the end, that'd probably be good, because we had timeout configurations in xapi that were sometimes in seconds, sometimes in milliseconds, so this should make it clearer.

Reply:
> The timeout is present in a configuration file, so we need to convert from a string,

But that is already taken care of by `Arg.Float` or `Arg.Int`

> and due to how config file parsing is implemented in XAPI we must always provide a conversion to a string too.

What is the to_string used for, exactly? Why is roundtripping being tested? What's stopping the printing from being defined as `Fmt.to_to_string Mtime.Span.pp`?

Reply:
Xcp_service.configure takes both a function that parses a string, and one that converts it back (e.g. to print in debug logs, or to print a configuration dump). That should be in a format that you can take and put into a config file, so it needs to be able to parse it back.
I could use Mtime.Span.pp, but there is no equivalent from_string function that can parse the pp output (the format of 'pp' is not defined on purpose, it is a pretty-printer, not a serializer).

*If* I can make all these timestamp be just simple floats expressed in seconds then none of this is needed (but I think there are some configuration values in XAPI where the values are not in seconds, but perhaps I can rename them and change them to be in seconds for consistency).

Reply:
Looking at the actual usage of Timer/Timeout so far it looks like what I was trying to protect against (accidentally using another value of type Mtime.Span.t instead of timeout) is a bit unlikely, so maybe I can drop this module (Timeout) and keep just Timer.

Reply:
After rebasing on master there is now only an Mtime.Span.t, and no Timeout anymore.